src/Avalonia.Controls/Primitives/ToggleButton.cs

```diff
@@ -37,7 +48,7 @@ protected override void OnClick()
 
         protected virtual void Toggle()
         {
-            IsChecked = !IsChecked;
+            IsChecked = IsChecked.HasValue ? (IsChecked.Value ? (IsThreeState ? (bool?)null : false) : true) : false;
```

Comment:
Can you expand this into if-else statements? It's hard to follow the logic of nested ternary statements.